audformat/core/column.py

```diff
@@ -338,6 +338,14 @@ def set(
             if is_scalar(values):
                 values = [values] * len(index)
             values = to_array(values)
+            if dtype == 'datetime64[ns]':
+                # Ensure all date values are timezone unaware,
+                # see https://github.com/audeering/audformat/issues/364
+                values = [
+                    pd.to_datetime(value).tz_localize(None)
```

Comment:
We need to convert to `pandas` datetime here as `numpy.datetime64` values don't provide the `tz_localize()` method.